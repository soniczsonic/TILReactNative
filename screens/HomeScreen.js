import React, {Component} from 'react';
import {
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  FlatList,
  StatusBar,
  SafeAreaView
} from 'react-native'
import {getAllUserTil} from '../fireStore/ORM'
import TilListItem from '../components/TilListItem'
import {NavigationEvents} from 'react-navigation';
import Layout from '../constants/Layout'
import Colors from '../constants/Colors'
import ActivityIndicator from '../components/ActivityIndicator'

export default class HomeScreen extends Component {
  static navigationOptions = {
    header: null
  }

  constructor(props) {
    super(props);
    this.state = {
      til: null,
      isLoading: true
    }
  }

  componentDidMount = async() => {
    this.loadFeed()
  }

  loadFeed = async() => {
    const til = await getAllUserTil()
    this.setState({til, isLoading: false})
  }

  render() {
    if (this.state.isLoading) return <ActivityIndicator />
    const { til } = this.state
    return (
      <SafeAreaView style={styles.safeAreaView}>
      <View style={styles.container}>
        <StatusBar
          backgroundColor="red"
          barStyle="light-content"
        />
        <View style={styles.heading}>
          <Text style={styles.headingTest}>Today I Learned</Text>
        </View>

        <TilListItem til={til} />

        <TouchableOpacity onPress={() => this.props.navigation.navigate('PostTilScreen')} style={styles.fab}>
          <Text style={styles.fabIcon}>+</Text>
        </TouchableOpacity>
        <NavigationEvents onDidFocus={() => this.loadFeed()} />

      </View>
      </SafeAreaView>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.tintColor
  },
  // safeAreaViewのstyleにflex: 1を加えたら治った。
  safeAreaView: {
    flex: 1,
    backgroundColor: Colors.tintColor
  },
  heading: {
    height: 60,
    backgroundColor: Colors.backgroundBaseColor,
    alignItems: 'center',
    justifyContent: 'center',
  },
  headingTest: {
    fontSize: 20,
    color: 'white',
    fontWeight: 'bold',
  },
  list: {
    margin: 5,
    backgroundColor: 'white',
    height:80,
    justifyContent: 'space-around',
    paddingLeft: 10,
    elevation: 1
  },
  fab: {
    position: 'absolute',
    width: 56,
    height: 56,
    alignItems: 'center',
    justifyContent: 'center',
    right: 20,
    bottom: 20,
    backgroundColor: '#03A9F4',
    borderRadius: 30,
    elevation: 8,
    zIndex: 2
  },
  fabIcon: {
    fontSize: 40,
    color: 'white'
  }
});
