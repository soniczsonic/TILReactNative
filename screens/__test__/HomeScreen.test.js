import React from 'react'
import HomeScreen from '../HomeScreen'
import renderer from 'react-test-renderer'

// Ioniconsは、<Text />と判定されるので、このテストに意味はないが、一応記述
it('not focused home icon', async () => {
  const tree = await renderer.create(<HomeScreen />).toJSON()

  expect(tree).toMatchSnapshot()
})
