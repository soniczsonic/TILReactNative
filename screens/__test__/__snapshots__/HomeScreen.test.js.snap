// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not focused home icon 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#03A9F4",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    animating={true}
    animationDuration={1200}
    animationEasing={[Function]}
    color="white"
    interaction={true}
    renderComponent={[Function]}
    size={50}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "height": 50,
          "marginHorizontal": 5,
          "width": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "height": 25,
            "transform": Array [
              Object {
                "translateY": 10,
              },
            ],
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "height": 25,
            "transform": Array [
              Object {
                "translateY": -10,
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 50,
          "marginHorizontal": 5,
          "width": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "height": 25,
            "transform": Array [
              Object {
                "translateY": 7.071067811865475,
              },
            ],
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "height": 25,
            "transform": Array [
              Object {
                "translateY": -7.071067811865475,
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 50,
          "marginHorizontal": 5,
          "width": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "height": 25,
            "transform": Array [
              Object {
                "translateY": 6.123233995736766e-16,
              },
            ],
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "height": 25,
            "transform": Array [
              Object {
                "translateY": -6.123233995736766e-16,
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 50,
          "marginHorizontal": 5,
          "width": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "height": 25,
            "transform": Array [
              Object {
                "translateY": 7.0710678118654755,
              },
            ],
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "height": 25,
            "transform": Array [
              Object {
                "translateY": -7.0710678118654755,
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 50,
          "marginHorizontal": 5,
          "width": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "height": 25,
            "transform": Array [
              Object {
                "translateY": 10,
              },
            ],
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "height": 25,
            "transform": Array [
              Object {
                "translateY": -10,
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;
